core/tests/webdriverio_desktop/diagnosticTestPage.js

```diff
@@ -0,0 +1,152 @@
+var AdminPage = require('../webdriverio_utils/AdminPage.js');
+var ClassroomPage = require('../webdriverio_utils/ClassroomPage.js');
+var DiagnosticTestPage = require('../webdriverio_utils/DiagnosticTestPage');
+var ExplorationPlayerPage = require(
+  '../webdriverio_utils/ExplorationPlayerPage.js');
+var SkillEditorPage = require('../webdriverio_utils/SkillEditorPage.js');
+var TopicsAndSkillsDashboardPage = require(
+  '../webdriverio_utils/TopicsAndSkillsDashboardPage.js');
+var TopicEditorPage = require('../webdriverio_utils/TopicEditorPage.js');
+
+describe('Diagnostic test page functionality', function() {
+  var adminPage = null;
+  var classroomPage = null;
+  var diagnosticTestPage = null;
+  var explorationPlayerPage = null;
+  var skillEditorPage = null;
+  var topicsAndSkillsDashboardPage = null;
+  var topicEditorPage = null;
+
+  beforeAll(async function() {
+    adminPage = new AdminPage.AdminPage();
+    classroomPage = new ClassroomPage.ClassroomPage();
+    explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+    diagnosticTestPage = new DiagnosticTestPage.DiagnosticTestPage();
+    skillEditorPage = new SkillEditorPage.SkillEditorPage();
+    topicsAndSkillsDashboardPage = (
+      new TopicsAndSkillsDashboardPage.TopicsAndSkillsDashboardPage());
+    topicEditorPage = (
+      new TopicEditorPage.TopicEditorPage());
+
+    await users.createAndLoginCurriculumAdminUser(
+      'creator@classroomPage.com', 'creatorClassroomPage');
+    var handle = await browser.getWindowHandle();
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.createTopic(
+      'Topic 1', 'topic-one', 'Description', false);
+    await topicEditorPage.submitTopicThumbnail('../data/test2_svg.svg', true);
+    await topicEditorPage.updateMetaTagContent('topic meta tag');
+    await topicEditorPage.updatePageTitleFragment('topic page title');
+    await topicEditorPage.saveTopic('Added thumbnail.');
+    var url = await browser.getUrl();
+    var topicId = url.split('/')[4].slice(0, -1);
+    await general.closeCurrentTabAndSwitchTo(handle);
+    await adminPage.editConfigProperty(
+      'The details for each classroom page.',
+      'List',
+      async function(elem) {
+        elem = await elem.editItem(0, 'Dictionary');
+        elem = await elem.editEntry(4, 'List');
+        elem = await elem.addItem('Unicode');
+        await action.setValue(
+          'Topic ID', elem, topicId, false);
+      });
+    await classroomPage.get('math');
+    // Even if the topic is unpublished, an unclickable tile is shown
+    // currently.
+    await classroomPage.expectNumberOfTopicsToBe(1);
+    await topicsAndSkillsDashboardPage.get();
+    (
+      await
+      topicsAndSkillsDashboardPage.createSkillWithDescriptionAndExplanation(
+        'Skill 1', 'Concept card explanation', false));
+    await skillEditorPage.addRubricExplanationForDifficulty(
+      'Easy', 'Second explanation for easy difficulty.');
```

Comment:
Why is this the "second" explanation? Where's the first?

Reply:
Updated